### ISSUE TITLE
CI: long-lived connection test failed

### ISSUE THREAD

**Issue by ianvernon** [cilium/cilium#5813]
Build link: https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/1727/testReport/junit/k8s-1/8/K8sChaosTest_Restart_with_long_lived_connections_TCP_connection_is_not_dropped_when_cilium_restarts/

Output:

```
Stacktrace

/home/jenkins/workspace/cilium-ginkgo_cilium_master-QVKDPVJ2A6E272IRUX7X6DCB5FBLCQDKRTOMHPZS7ZECX3RM5C3A/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:383
Failed while cilium was restarting
Expected command: kubectl exec -n default netperf-client -- netperf -l 300 -t TCP_STREAM -H 10.10.0.173 
To succeed, but it failed:
Exitcode: 1 
Stdout:
 	 MIGRATED TCP STREAM TEST from 0.0.0.0 (0.0.0.0) port 0 AF_INET to 10.10.0.173 (10.10.0) port 0 AF_INET
	 recv_response_timed_n: no response received. errno 104 counter 0
	 command terminated with exit code 1
	 
Stderr:
 	 

/home/jenkins/workspace/cilium-ginkgo_cilium_master-QVKDPVJ2A6E272IRUX7X6DCB5FBLCQDKRTOMHPZS7ZECX3RM5C3A/src/github.com/cilium/cilium/test/k8sT/Chaos.go:247
Standard Output

Number of 'context deadline exceeded' in logs: 0
Number of 'level=error' in logs: 0
Number of 'level=warning' in logs: 2
Number of 'Cilium API handler panicked' in logs: 0
⚠️  Number of 'Goroutine took lock for more than' in logs: 25
Cilium pods: [cilium-4kgsh cilium-p84mc]
Netpols loaded: 
CiliumNetworkPolicies loaded: 
Endpoint Policy Enforcement:
     kube-dns-67c6cf67d7-kqhfr          =>   none
     prometheus-core-5c9fbf5cc6-jpfpw   =>   none
     netperf-client                     =>   none
     netperf-server                     =>   none
     cilium-etcd-bdhd5hx7ck             =>   none
     cilium-etcd-k277j9kpqj             =>   none
     cilium-etcd-sw2j9rzhj9             =>   none
     etcd-operator-bbdc658cf-sl4wp      =>   none
Cilium agent 'cilium-4kgsh': Status: Failure  Health: Warning Nodes "k8s1 k8s2" ContinerRuntime: Ok Kubernetes: Ok KVstore: Failure Controllers: Total 29 Failed 0
⚠️  Cilium agent 'cilium-p84mc': Status: Ok  Health: Ok Nodes "k8s2" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 46 Failed 1
Failed controllers:
 controller kvstore-sync-store-cilium/state/nodes/v1 failure 'etcdserver: request timed out'

Standard Error

STEP: Deleting all cilium pods
STEP: Waiting cilium pods to terminate
STEP: Waiting for cilium pods to be ready
STEP: Stopping netperf client test
=== Test Finished at 2018-10-05T10:50:21Z====
===================== TEST FAILED =====================
cmd: kubectl get pods -o wide --all-namespaces
Exitcode: 0 
Stdout:
 	 NAMESPACE     NAME                               READY     STATUS    RESTARTS   AGE       IP              NODE
	 default       netperf-client                     1/1       Running   0          4m        10.10.1.9       k8s2
	 default       netperf-server                     1/1       Running   0          4m        10.10.0.173     k8s1
	 kube-system   cilium-4kgsh                       1/1       Running   1          3m        192.168.36.11   k8s1
	 kube-system   cilium-etcd-bdhd5hx7ck             1/1       Running   0          14m       10.10.0.74      k8s1
	 kube-system   cilium-etcd-k277j9kpqj             1/1       Running   0          14m       10.10.0.134     k8s1
	 kube-system   cilium-etcd-sw2j9rzhj9             0/1       Running   0          8m        10.10.1.189     k8s2
	 kube-system   cilium-p84mc                       1/1       Running   0          3m        192.168.36.12   k8s2
	 kube-system   etcd-k8s1                          1/1       Running   0          22m       192.168.36.11   k8s1
	 kube-system   etcd-operator-bbdc658cf-sl4wp      1/1       Running   0          15m       10.10.1.25      k8s2
	 kube-system   kube-apiserver-k8s1                1/1       Running   0          21m       192.168.36.11   k8s1
	 kube-system   kube-controller-manager-k8s1       1/1       Running   0          21m       192.168.36.11   k8s1
	 kube-system   kube-dns-67c6cf67d7-kqhfr          3/3       Running   0          22m       10.10.1.167     k8s2
	 kube-system   kube-proxy-vwtk2                   1/1       Running   0          22m       192.168.36.11   k8s1
	 kube-system   kube-proxy-vx5ks                   1/1       Running   0          15m       192.168.36.12   k8s2
	 kube-system   kube-scheduler-k8s1                1/1       Running   0          21m       192.168.36.11   k8s1
	 monitoring    prometheus-core-5c9fbf5cc6-jpfpw   1/1       Running   0          15m       10.10.1.32      k8s2
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-4kgsh -- cilium service list
Exitcode: 0 
Stdout:
 	 ID   Frontend              Backend                   
	 1    10.96.0.10:53         1 => 10.10.1.167:53       
	 2    10.96.0.1:443         1 => 192.168.36.11:6443   
	 3    10.109.222.88:12865                             
	 4    10.104.195.75:9090    1 => 10.10.1.32:9090      
	 5    10.107.200.71:2379    1 => 10.10.1.189:2379     
	                            2 => 10.10.0.134:2379     
	                            3 => 10.10.0.74:2379      
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-4kgsh -- cilium endpoint list
Exitcode: 0 
Stdout:
 	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                       IPv6                 IPv4          STATUS   
	            ENFORCEMENT        ENFORCEMENT                                                                                                       
	 2403       Disabled           Disabled          128        container:annotation.etcd.version=3.3.9           f00d::a0a:0:0:963    10.10.0.134   ready       
	                                                            k8s:app=etcd                                                                                     
	                                                            k8s:etcd_cluster=cilium-etcd                                                                     
	                                                            k8s:etcd_node=cilium-etcd-k277j9kpqj                                                             
	                                                            k8s:io.cilium.fixed-identity=kv-store                                                            
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                         
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                  
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                      
	 26786      Disabled           Disabled          4          reserved:health                                   f00d::a0a:0:0:68a2   10.10.0.190   ready       
	 51439      Disabled           Disabled          128        container:annotation.etcd.version=3.3.9           f00d::a0a:0:0:c8ef   10.10.0.74    ready       
	                                                            k8s:app=etcd                                                                                     
	                                                            k8s:etcd_cluster=cilium-etcd                                                                     
	                                                            k8s:etcd_node=cilium-etcd-bdhd5hx7ck                                                             
	                                                            k8s:io.cilium.fixed-identity=kv-store                                                            
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                         
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                  
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                      
	 64157      Disabled           Disabled          61899      k8s:id=netperf-server                             f00d::a0a:0:0:fa9d   10.10.0.173   restoring   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                         
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                  
	                                                            k8s:io.kubernetes.pod.namespace=default                                                          
	                                                            k8s:zgroup=testapp                                                                               
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-p84mc -- cilium service list
Exitcode: 0 
Stdout:
 	 ID   Frontend              Backend                   
	 1    10.96.0.1:443         1 => 192.168.36.11:6443   
	 2    10.109.222.88:12865                             
	 3    10.107.200.71:2379    1 => 10.10.0.134:2379     
	                            2 => 10.10.0.74:2379      
	                            3 => 10.10.1.189:2379     
	 4    10.96.0.10:53         1 => 10.10.1.167:53       
	 5    10.104.195.75:9090    1 => 10.10.1.32:9090      
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-p84mc -- cilium endpoint list
Exitcode: 0 
Stdout:
 	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                              IPv6                   IPv4          STATUS   
	            ENFORCEMENT        ENFORCEMENT                                                                                                                
	 19103      Disabled           Disabled          4          reserved:health                                          f00d::a0a:100:0:4a9f   10.10.1.38    ready   
	 29554      Disabled           Disabled          129        k8s:io.cilium.fixed-identity=kube-dns                    f00d::a0a:100:0:7372   10.10.1.167   ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=kube-dns                                                      
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
	                                                            k8s:k8s-app=kube-dns                                                                                  
	 43856      Disabled           Disabled          9189       k8s:app=prometheus                                       f00d::a0a:100:0:ab50   10.10.1.32    ready   
	                                                            k8s:component=core                                                                                    
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=prometheus-k8s                                                
	                                                            k8s:io.kubernetes.pod.namespace=monitoring                                                            
	 47309      Disabled           Disabled          128        container:annotation.etcd.version=3.3.9                  f00d::a0a:100:0:b8cd   10.10.1.189   ready   
	                                                            k8s:app=etcd                                                                                          
	                                                            k8s:etcd_cluster=cilium-etcd                                                                          
	                                                            k8s:etcd_node=cilium-etcd-sw2j9rzhj9                                                                  
	                                                            k8s:io.cilium.fixed-identity=kv-store                                                                 
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
	 52947      Disabled           Disabled          26736      k8s:id=netperf-client                                    f00d::a0a:100:0:ced3   10.10.1.9     ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
	                                                            k8s:io.kubernetes.pod.namespace=default                                                               
	                                                            k8s:zgroup=testapp                                                                                    
	 60652      Disabled           Disabled          128        k8s:io.cilium.fixed-identity=kv-store                    f00d::a0a:100:0:ecec   10.10.1.25    ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=cilium-etcd-sa                                                
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
	                                                            k8s:name=etcd-operator                                                                                
	 
Stderr:
 	 

===================== Existing AfterFailed =====================
Something has failed on AfterEach, running AfterFailed functions
Something has failed on AfterEach, running AfterFailed functions
```

Logs: [98b25332_K8sChaosTest_Restart_with_long_lived_connections_TCP_connection_is_not_dropped_when_cilium_restarts.zip](https://github.com/cilium/cilium/files/2451471/98b25332_K8sChaosTest_Restart_with_long_lived_connections_TCP_connection_is_not_dropped_when_cilium_restarts.zip)

**Comment by ianvernon** [comment]
Assigning myself to triage since I am Jenkins hat.

**Comment by ianvernon** [comment]
One of the cilium-etcd pods is Running, but not Ready:

```
10:50:23  	 NAMESPACE     NAME                               READY     STATUS    RESTARTS   AGE       IP              NODE
10:50:23 	 default       netperf-client                     1/1       Running   0          4m        10.10.1.9       k8s2
10:50:23 	 default       netperf-server                     1/1       Running   0          4m        10.10.0.173     k8s1
10:50:23 	 kube-system   cilium-4kgsh                       1/1       Running   1          3m        192.168.36.11   k8s1
10:50:23 	 kube-system   cilium-etcd-bdhd5hx7ck             1/1       Running   0          14m       10.10.0.74      k8s1
10:50:23 	 kube-system   cilium-etcd-k277j9kpqj             1/1       Running   0          14m       10.10.0.134     k8s1
10:50:23 	 kube-system   cilium-etcd-sw2j9rzhj9             0/1       Running   0          8m        10.10.1.189     k8s2
10:50:23 	 kube-system   cilium-p84mc                       1/1       Running   0          3m        192.168.36.12   k8s2
10:50:23 	 kube-system   etcd-k8s1                          1/1       Running   0          22m       192.168.36.11   k8s1
10:50:23 	 kube-system   etcd-operator-bbdc658cf-sl4wp      1/1       Running   0          15m       10.10.1.25      k8s2
10:50:23 	 kube-system   kube-apiserver-k8s1                1/1       Running   0          21m       192.168.36.11   k8s1
10:50:23 	 kube-system   kube-controller-manager-k8s1       1/1       Running   0          21m       192.168.36.11   k8s1
10:50:23 	 kube-system   kube-dns-67c6cf67d7-kqhfr          3/3       Running   0          22m       10.10.1.167     k8s2
10:50:23 	 kube-system   kube-proxy-vwtk2                   1/1       Running   0          22m       192.168.36.11   k8s1
10:50:23 	 kube-system   kube-proxy-vx5ks                   1/1       Running   0          15m       192.168.36.12   k8s2
10:50:23 	 kube-system   kube-scheduler-k8s1                1/1       Running   0          21m       192.168.36.11   k8s1
10:50:23 	 monitoring    prometheus-core-5c9fbf5cc6-jpfpw   1/1       Running   0          15m       10.10.1.32      k8s2
```

**Comment by ianvernon** [comment]
`test-output.log` shows that the netperf command was ran around the following:

```
time="2018-10-05T10:49:41Z" level=info msg="WaitforPods: pods are not ready" data="false false" filter="-l k8s-app=cilium" namespace=kube-system
cmd: "kubectl exec -n default netperf-client -- netperf -l 300 -t TCP_STREAM -H 10.10.0.173" exitCode: 1 duration: 3m37.58908213s stdout:
MIGRATED TCP STREAM TEST from 0.0.0.0 (0.0.0.0) port 0 AF_INET to 10.10.0.173 (10.10.0) port 0 AF_INET
recv_response_timed_n: no response received. errno 104 counter 0
command terminated with exit code 1

time="2018-10-05T10:49:46Z" level=debug msg="running command: kubectl -n kube-system get pods -l k8s-app=cilium -o json"
```

**Comment by ianvernon** [comment]
The logs from etcd show that there were definitively connectivity issues for it, for `pod-kube-system-cilium-etcd-bdhd5hx7ck-etcd.log`:

```
2018-10-05T10:42:19.93215932Z 2018-10-05 10:42:19.931923 I | rafthttp: peer dceaa53c621e1ed became active
2018-10-05T10:42:19.932188967Z 2018-10-05 10:42:19.931956 I | rafthttp: established a TCP streaming connection with peer dceaa53c621e1ed (stream MsgApp v2 reader)
2018-10-05T10:42:19.932571699Z 2018-10-05 10:42:19.932493 I | rafthttp: established a TCP streaming connection with peer dceaa53c621e1ed (stream Message reader)
2018-10-05T10:42:19.952113987Z 2018-10-05 10:42:19.952020 I | rafthttp: established a TCP streaming connection with peer dceaa53c621e1ed (stream Message writer)
2018-10-05T10:42:19.959683516Z 2018-10-05 10:42:19.959275 I | rafthttp: established a TCP streaming connection with peer dceaa53c621e1ed (stream MsgApp v2 writer)
2018-10-05T10:42:20.550073986Z 2018-10-05 10:42:20.549979 W | rafthttp: health check for peer dceaa53c621e1ed could not connect: dial tcp 10.10.1.189:2380: connect: connection refused
2018-10-05T10:47:46.591174531Z 2018-10-05 10:47:46.590780 W | rafthttp: lost the TCP streaming connection with peer dceaa53c621e1ed (stream MsgApp v2 reader)
2018-10-05T10:47:46.591195025Z 2018-10-05 10:47:46.590857 E | rafthttp: failed to read dceaa53c621e1ed on stream MsgApp v2 (read tcp 10.10.0.74:37714->10.10.1.189:2380: i/o timeout)
2018-10-05T10:47:46.591209Z 2018-10-05 10:47:46.590861 I | rafthttp: peer dceaa53c621e1ed became inactive
2018-10-05T10:47:46.863155326Z 2018-10-05 10:47:46.862753 W | rafthttp: lost the TCP streaming connection with peer dceaa53c621e1ed (stream Message reader)
2018-10-05T10:48:53.938934002Z 2018-10-05 10:48:53.938823 W | rafthttp: lost the TCP streaming connection with peer dceaa53c621e1ed (stream Message writer)
2018-10-05T10:52:15.095922386Z 2018-10-05 10:52:15.095844 I | etcdserver/membership: removed member dceaa53c621e1ed from cluster 2d55edc1ae8595e5
2018-10-05T10:52:15.095935258Z 2018-10-05 10:52:15.095878 I | rafthttp: stopping peer dceaa53c621e1ed...
2018-10-05T10:52:15.096268216Z 2018-10-05 10:52:15.096211 I | rafthttp: closed the TCP streaming connection with peer dceaa53c621e1ed (stream MsgApp v2 writer)
2018-10-05T10:52:15.097172509Z 2018-10-05 10:52:15.097099 I | rafthttp: stopped streaming with peer dceaa53c621e1ed (writer)
2018-10-05T10:52:15.097179368Z 2018-10-05 10:52:15.097118 I | rafthttp: stopped streaming with peer dceaa53c621e1ed (writer)
2018-10-05T10:52:15.118097909Z 2018-10-05 10:52:15.117946 I | rafthttp: stopped HTTP pipelining with peer dceaa53c621e1ed
2018-10-05T10:52:15.118120835Z 2018-10-05 10:52:15.117996 I | rafthttp: stopped streaming with peer dceaa53c621e1ed (stream MsgApp v2 reader)
2018-10-05T10:52:15.118125262Z 2018-10-05 10:52:15.118009 I | rafthttp: stopped streaming with peer dceaa53c621e1ed (stream Message reader)
2018-10-05T10:52:15.118128796Z 2018-10-05 10:52:15.118013 I | rafthttp: stopped peer dceaa53c621e1ed
2018-10-05T10:52:15.118132257Z 2018-10-05 10:52:15.118020 I | rafthttp: removed peer dceaa53c621e1ed
```

**Comment by ianvernon** [comment]
I'm guessing where the following was ran was right before the netperf command which failed was started:
```
STEP: Deleting all cilium pods
time="2018-10-05T10:46:05Z" level=debug msg="running command: kubectl -n kube-system delete pods -l k8s-app=cilium"
```

**Comment by ianvernon** [comment]
I was hoping monitor logs would reveal issues as to why the connectivity check failed, but we have no monitor logs. Will file an issue for that.

**Comment by ianvernon** [comment]
Not sure what else I can do here to debug since we have no visibility into the datapath. I'll keep this issue around in case this reproduces in a future build with more logs.

**Comment by tgraf** [comment]
The test is flawed, the data connection port is randomly chosen in the control connection and cannot be load-balanced in the way designed by the test.

I suggest to use iperf.

EDIT: The test has code to retrieve the serviceIP but it does not seem to use it. It is talking to the pod IP directly which should work.

**Comment by aanm** [comment]
[ec25572d_K8sChaosTest_Restart_with_long_lived_connections_TCP_connection_is_not_dropped_when_cilium_restarts.zip](https://github.com/cilium/cilium/files/2456412/ec25572d_K8sChaosTest_Restart_with_long_lived_connections_TCP_connection_is_not_dropped_when_cilium_restarts.zip)

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/1017

**Comment by aanm** [comment]
It happen again... Increasing to priority/release-blocker

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/1023/

[345fa83c_K8sChaosTest_Restart_with_long_lived_connections_L3-L4_policies_still_work_while_Cilium_is_restarted.zip](https://github.com/cilium/cilium/files/2458299/345fa83c_K8sChaosTest_Restart_with_long_lived_connections_L3-L4_policies_still_work_while_Cilium_is_restarted.zip)

**Comment by tgraf** [comment]
kube-dns was not able to be restored:
```
2018-10-08T21:38:44.941284177Z level=warning msg="clang-3.8: error: clang frontend command failed due to signal (use -v to see invocation)" subsys=datapath-loader
2018-10-08T21:38:44.941310534Z level=warning msg="clang version 3.8.1 (tags/RELEASE_381/final)" subsys=datapath-loader
2018-10-08T21:38:44.941363029Z level=warning msg="Target: bpf" subsys=datapath-loader
2018-10-08T21:38:44.941370164Z level=warning msg="Thread model: posix" subsys=datapath-loader
2018-10-08T21:38:44.941399172Z level=warning msg="InstalledDir: /usr/local/clang+llvm/bin" subsys=datapath-loader
2018-10-08T21:38:44.941471387Z level=warning msg="clang-3.8: note: diagnostic msg: PLEASE submit a bug report to http://llvm.org/bugs/ and include the crash backtrace, preprocessed source, and associated run script." subsys=datapath-loader
2018-10-08T21:38:44.941480323Z level=warning msg="clang-3.8: note: diagnostic msg: " subsys=datapath-loader
2018-10-08T21:38:44.941531362Z level=warning msg="********************" subsys=datapath-loader
2018-10-08T21:38:44.941556264Z level=warning subsys=datapath-loader
2018-10-08T21:38:44.941616336Z level=warning msg="PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:" subsys=datapath-loader
2018-10-08T21:38:44.941634344Z level=warning msg="Preprocessed source(s) and associated run script(s) are located at:" subsys=datapath-loader
2018-10-08T21:38:44.941687461Z level=warning msg="clang-3.8: note: diagnostic msg: /tmp/bpf_lxc-647016.c" subsys=datapath-loader
2018-10-08T21:38:44.941705204Z level=warning msg="clang-3.8: note: diagnostic msg: /tmp/bpf_lxc-647016.sh" subsys=datapath-loader
2018-10-08T21:38:44.941758673Z level=warning msg="clang-3.8: note: diagnostic msg: " subsys=datapath-loader
2018-10-08T21:38:44.941767091Z level=warning subsys=datapath-loader
2018-10-08T21:38:44.941793055Z level=warning msg="********************" subsys=datapath-loader
2018-
```

Related to https://github.com/cilium/cilium/issues/5683

**Comment by ianvernon** [comment]
cilium-p2n8c was the pod managing kube-dns. I checked its bugtool folder for any information about killed processes from `dmesg`, and interestingly enough, the logs end right before the time from when `clang` was killed (`2018-10-08T21:38:44.941284177Z`).

The last logs from `dmesg` end a second before...:
```
...
2018-10-08T21:38:43,408840+0000 IPv6: ADDRCONF(NETDEV_UP): cilium_health: link is not ready
2018-10-08T21:38:43,522501+0000 IPVS: Creating netns size=2176 id=42
2018-10-08T21:38:43,585495+0000 IPv6: ADDRCONF(NETDEV_CHANGE): cilium_health: link becomes ready
```

The test itself ends later:

```
STEP: Stopping netperf client test
cmd: "kubectl exec -n default netperf-client -- netperf -l 300 -t TCP_STREAM -H 10.10.1.174" exitCode: 130 duration: 29.486026651s stdout:
MIGRATED TCP STREAM TEST from 0.0.0.0 (0.0.0.0) port 0 AF_INET to 10.10.1.174 (10.10.1) port 0 AF_INET
^C
=== Test Finished at 2018-10-08T21:38:50Z====
```

It is worrying to me that we can't even get logs from dmesg? :/ 

I think the right approach to debugging this in the absence of logs, is to see if there was anything running at the time that would correlate with clang getting killed. The test itself is running netperf in a pod in the background, alongside etcd and kube-dns:

```
time="2018-10-08T21:38:53Z" level=debug msg="running command: kubectl exec -n kube-system cilium-p2n8c -- cilium endpoint list"
cmd: "kubectl exec -n kube-system cilium-p2n8c -- cilium endpoint list" exitCode: 0 duration: 356.011382ms
cmd: kubectl exec -n kube-system cilium-p2n8c -- cilium endpoint list
Exitcode: 0 
Stdout:
 	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                              IPv6                   IPv4          STATUS   
	            ENFORCEMENT        ENFORCEMENT                                                                                                                
	 20667      Disabled           Disabled          128        container:annotation.etcd.version=3.3.9                  f00d::a0a:100:0:50bb   10.10.1.127   ready       
	                                                            k8s:app=etcd                                                                                              
	                                                            k8s:etcd_cluster=cilium-etcd                                                                              
	                                                            k8s:etcd_node=cilium-etcd-2b969zv757                                                                      
	                                                            k8s:io.cilium.fixed-identity=kv-store                                                                     
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                  
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                           
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                               
	 35114      Enabled            Disabled          9219       k8s:id=netperf-server                                    f00d::a0a:100:0:892a   10.10.1.174   ready       
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                  
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                           
	                                                            k8s:io.kubernetes.pod.namespace=default                                                                   
	                                                            k8s:zgroup=testapp                                                                                        
	 44181      Disabled           Disabled          129        k8s:io.cilium.fixed-identity=kube-dns                    f00d::a0a:100:0:ac95   10.10.1.156   not-ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                  
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=coredns                                                           
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                               
	                                                            k8s:k8s-app=kube-dns                                                                                      
	 52031      Disabled           Disabled          4          reserved:health                                          f00d::a0a:100:0:cb3f   10.10.1.208   ready       
	 58130      Disabled           Disabled          128        k8s:io.cilium.fixed-identity=kv-store                    f00d::a0a:100:0:e312   10.10.1.192   ready       
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                  
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=cilium-etcd-sa                                                    
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                               
	                                                            k8s:name=etcd-operator      
```

Are these VMs able to handle this amount of load? 

<details>
<summary>
Logs containing `loadinfo` before 21:38:50 from `cilium-p2n8c` don't appear to reveal much
</summary>
```
$ grep "loadinfo" /Users/ianvernon/Downloads/test_results-3/111-/K8sChaosTest_Restart_with_long_lived_connections_L3-L4_policies_still_work_while_Cilium_is_restarted/pod-kube-system-cilium-p2n8c-cilium-agent.log 
2018-10-08T21:38:37.268461755Z level=debug msg="Load 1-min: 0.65 5-min: 0.80 15min: 0.79" subsys=loadinfo type=background
2018-10-08T21:38:37.268607248Z level=debug msg="Memory: Total: 3949 Used: 949 (24.04%) Free: 675 Buffers: 84 Cached: 2240" subsys=loadinfo type=background
2018-10-08T21:38:37.268754264Z level=debug msg="Swap: Total: 0 Used: 0 (0.00%) Free: 0" subsys=loadinfo type=background
2018-10-08T21:38:37.26984989Z level=debug msg="NAME cilium-agent STATUS R PID 1 CPU: 21.27% MEM: 1.62% CMDLINE: cilium-agent --debug=true -t=vxlan --kvstore=etcd --kvstore-opt=etcd.config=/var/lib/etcd-config/etcd.config --disable-ipv4=false --k8s-require-ipv4-pod-cidr --pprof=true --log-system-load --fixed-identity-mapping=128=kv-store --fixed-identity-mapping=129=kube-dns MEM-EXT: RSS: 63 VMS: 751 Data: 0 Stack: 0 Locked: 0 Swap: 0" subsys=loadinfo type=background
2018-10-08T21:38:42.27109696Z level=debug msg="Load 1-min: 0.76 5-min: 0.82 15min: 0.80" subsys=loadinfo type=background
2018-10-08T21:38:42.271277411Z level=debug msg="Memory: Total: 3949 Used: 1086 (27.51%) Free: 537 Buffers: 84 Cached: 2240" subsys=loadinfo type=background
2018-10-08T21:38:42.271397792Z level=debug msg="Swap: Total: 0 Used: 0 (0.00%) Free: 0" subsys=loadinfo type=background
2018-10-08T21:38:42.271718267Z level=debug msg="NAME cilium-agent STATUS S PID 1 CPU: 8.45% MEM: 2.11% CMDLINE: cilium-agent --debug=true -t=vxlan --kvstore=etcd --kvstore-opt=etcd.config=/var/lib/etcd-config/etcd.config --disable-ipv4=false --k8s-require-ipv4-pod-cidr --pprof=true --log-system-load --fixed-identity-mapping=128=kv-store --fixed-identity-mapping=129=kube-dns MEM-EXT: RSS: 83 VMS: 782 Data: 0 Stack: 0 Locked: 0 Swap: 0" subsys=loadinfo type=background
2018-10-08T21:38:47.273975584Z level=debug msg="Load 1-min: 0.86 5-min: 0.84 15min: 0.81" subsys=loadinfo type=background
2018-10-08T21:38:47.274202696Z level=debug msg="Memory: Total: 3949 Used: 1137 (28.79%) Free: 485 Buffers: 84 Cached: 2241" subsys=loadinfo type=background
2018-10-08T21:38:47.274391075Z level=debug msg="Swap: Total: 0 Used: 0 (0.00%) Free: 0" subsys=loadinfo type=background
2018-10-08T21:38:47.274860126Z level=debug msg="NAME cilium-agent STATUS S PID 1 CPU: 8.51% MEM: 2.67% CMDLINE: cilium-agent --debug=true -t=vxlan --kvstore=etcd --kvstore-opt=etcd.config=/var/lib/etcd-config/etcd.config --disable-ipv4=false --k8s-require-ipv4-pod-cidr --pprof=true --log-system-load --fixed-identity-mapping=128=kv-store --fixed-identity-mapping=129=kube-dns MEM-EXT: RSS: 105 VMS: 1017 Data: 0 Stack: 0 Locked: 0 Swap: 0" subsys=loadinfo type=background
2018-10-08T21:38:47.275250765Z level=debug msg="NAME cilium-node-monitor STATUS S PID 278 CPU: 2.57% MEM: 0.45% CMDLINE: cilium-node-monitor --bpf-root /sys/fs/bpf MEM-EXT: RSS: 17 VMS: 536 Data: 0 Stack: 0 Locked: 0 Swap: 0" subsys=loadinfo type=background
2018-10-08T21:38:47.275626861Z level=debug msg="NAME cilium-health STATUS S PID 313 CPU: 4.83% MEM: 0.43% CMDLINE: cilium-health -d --admin=unix --passive --pidfile /var/run/cilium/state/health-endpoint.pid MEM-EXT: RSS: 16 VMS: 395 Data: 0 Stack: 0 Locked: 0 Swap: 0" subsys=loadinfo type=background
2018-10-08T21:38:47.27596501Z level=debug msg="NAME cilium-health STATUS S PID 319 CPU: 6.15% MEM: 0.44% CMDLINE: cilium-health -d MEM-EXT: RSS: 17 VMS: 612 Data: 0 Stack: 0 Locked: 0 Swap: 0" subsys=loadinfo type=background
2018-10-08T21:38:47.276358337Z level=debug msg="NAME llc STATUS R PID 338 CPU: 8.62% MEM: 0.65% CMDLINE: llc -march=bpf -mcpu=probe -filetype=obj -o 20667_next/bpf_lxc.o MEM-EXT: RSS: 25 VMS: 58 Data: 0 Stack: 0 Locked: 0 Swap: 0" subsys=loadinfo type=background
```
</details>

**Comment by tgraf** [comment]
Given clang was killed, this is a duplicate of #5748 and fixed by #5837
